examples/mnist.cpp

```diff
@@ -110,29 +111,29 @@ void testMNIST() {
   auto &mod = EE.getModule();
   Function *F = mod.createFunction("main");
 
-  Variable *A =
-      mod.createVariable(ElemKind::FloatTy, {minibatchSize, 28, 28, 1}, "input",
-                         VisibilityKind::Public, false);
+  Placeholder *A = mod.createPlaceholder(
+      ElemKind::FloatTy, {minibatchSize, 28, 28, 1}, "input", false);
 
-  auto *CV0 = F->createConv("conv", A, 16, 5, 1, 2, 1);
+  auto *CV0 = F->createConv(ctx, "conv", A, 16, 5, 1, 2, 1);
```

Comment:
The issue is not the allocation but the initialization of weights. Inside the Conv builder we need to initialize some tensor with Xavier or other init method. I think that a better API would be something similar to LLVM's IRBuilder. However, I prefer to finish the migration because switching the API together with converting the underlying implementation would just be hell. So, step 1 would be to convert the tetsts. step2 would be to clean up the API.

Reply:
Ah, I see how the ctx helps to differentiate which method to call, that's sad that we will have to come back to remove them again, but I don't see a way around it.

Nevermind me :)